tests/system_test.py

```diff
@@ -1453,6 +1488,14 @@ def get_log(self, limit=None):
             cmd += " limit=%s" % limit
         return json.loads(self(cmd))
 
+    def delete_all_entities(self, long_entity_type):
+        # First, query the router to get all entities of entity type
+        results = self.query(long_entity_type)
+        for result in results:
+            entity_name = result.get('name')
+            if entity_name:
+                self.delete(long_entity_type, name=entity_name)
+
```

Comment:
Since the delete command is synchronous, the delete is successful (meaning the entity is deleted) by the time the response to the delete command reaches qdmanage. So, does adding an additional retry loop buy us anything since we already know that the entity has already been deleted ?
If the delete command is going to start a core thread action, we never know when that is going to complete ?

Reply:
Also, there  is a loop that deletes all connections as well.